### PR TITLE
Cleanup and add more JSDoc to phoenix.js

### PR DESCRIPTION
**Pull request by davydog187**
The formatting of `phoenix.js` is inconsistent, and a lot of the code is missing JSDoc. This causes https://hexdocs.pm/phoenix/js/ to have a lot of missing, useful information.

This PR aims to take a stab at that problem :)

### CODE REVIEW

**Comment by davydog187** [comment]
Looks like this failed due to unrelated errors

File: assets/js/phoenix.js [hunk 1]
```diff
@@ -386,18 +412,42 @@ export class Channel {
...
   }
 
+  /**
+   * Hook into channel close
+   * @param {function} callback
+   */
   onClose(callback){ this.on(CHANNEL_EVENTS.close, callback) }
 
+  /**
+   * Hook into channel close
```
Comment (davydog187) [inline assets/js/phoenix.js]: channel error ***

**Comment by davydog187** [comment]
Resolved the merge conflicts.

File: assets/js/phoenix.js [hunk 2]
```diff
@@ -387,41 +413,65 @@ export class Channel {
...
-  *     ref2 = channel.on("event", do_other_stuff)
-  *     channel.off("event", ref1)
-  *     // Since unsubscription, do_stuff won't fire,
-  *     // while do_other_stuff will keep firing on the "event" 
-  * ```
-  */
+  /**
+   * Subscribes on channel events
+   *
+   * Subscription returnrs a ref counter, which can be used later to
```
Comment (davydog187) [inline assets/js/phoenix.js]: `returns`

File: assets/js/phoenix.js [hunk 3]
```diff
@@ -387,41 +413,65 @@ export class Channel {
...
-  *     channel.off("event", ref1)
-  *     // Since unsubscription, do_stuff won't fire,
-  *     // while do_other_stuff will keep firing on the "event" 
-  * ```
-  */
+  /**
+   * Subscribes on channel events
+   *
+   * Subscription returnrs a ref counter, which can be used later to 
+   * unsubsribe exact event listener:
```
Comment (davydog187) [inline assets/js/phoenix.js]: `unsubscribe` and add `the`, no `:`

File: assets/js/phoenix.js [hunk 4]
```diff
@@ -387,41 +413,65 @@ export class Channel {
...
+   *
+   * ```javascript
+   *     ref1 = channel.on("event", do_stuff)
```
Comment (davydog187) [inline assets/js/phoenix.js]: no extra indentation

**Comment by davydog187** [comment]
Hi @chrismccord, I will continue to work on this, unless you don't find this useful. Any feedback would be apprectiated

**Comment by chrismccord** [comment]
Hold tight! There's a previous linter PR that I rejected and you can see my rationale there. I have some other things to take care of before I can give this one proper consideration, so give me a bit. Thanks! :)

**Comment by davydog187** [comment]
Great, thank you! Just an FYI, this PR is just documentation. My other PR, #2418 deals with adding an eslint config

**Comment by chrismccord** [comment]
😍😍😍

**Pull request merged**